common/common_structs/src/wrapper_types.rs

```diff
@@ -92,3 +94,50 @@ impl<M: ManagedTypeApi> RawResultWrapper<M> {
         result
     }
 }
+
+pub struct PaymentsWrapper<M: ManagedTypeApi> {
+    payments: PaymentsVec<M>,
+}
+
+impl<M: ManagedTypeApi> PaymentsWrapper<M>
+where
+    M: ManagedTypeApi + SendApi,
+{
+    pub fn new() -> Self {
+        Self {
+            payments: ManagedVec::new(),
+        }
+    }
+
+    pub fn new_from_payments(payments: PaymentsVec<M>) -> Self {
+        Self { payments }
+    }
+
+    pub fn push(&mut self, payment: EsdtTokenPayment<M>) {
+        if payment.amount > 0 {
+            self.payments.push(payment);
+        }
+    }
+
+    pub fn send_to_address<T: SendApi>(&self, address: &ManagedAddress<M>) {
+        if !self.payments.is_empty() {
+            for payment in self.payments.iter() {
+                if payment.amount > 0 {
+                    let _ = T::send_api_impl().transfer_esdt_nft_execute(
+                        address,
+                        &payment.token_identifier,
+                        payment.token_nonce,
+                        &payment.amount,
+                        0,
+                        &ManagedBuffer::new(),
+                        &ManagedArgBuffer::new(),
+                    );
+                }
+            }
+        }
```

Comment:
Use multi-transfer here. Also, no need for `> 0` check, as you already do that when adding the payment.